### PR TITLE
New WebAssembly demo and updated roadmap

### PR DESCRIPTION
**Pull request by s3ththompson**
Included in this PR:

- Swapped out AngryBots for Tanks! demo (and updated screenshot)

  screen shot 2017-03-03 at 5 56 51 pm
- Removed instructions for turning on WebAssembly in experimental settings
- Dropped older WebAssembly feature detection (`0xa` through `0xd`)
- Updated roadmap page to include [cross-browser consensus](https://lists.w3.org/Archives/Public/public-webassembly/2017Feb/0002.html)
- Updated site banner to highlight cross-browser consensus
 
   screen shot 2017-03-03 at 5 55 06 pm
- Added logo selection and cross-browser consensus to past milestones section on roadmap page
- Added [MDN link](https://developer.mozilla.org/en-US/docs/WebAssembly) to left-nav of Getting Started pages

Deferred to future PR:

- Adding official logo (need to optimize assets for small sizes)
- Adding list of supported browsers (should probably highlight the Chrome/Firefox versions that support WebAssembly)
- Adding more prominent MDN link (should probably be more than a sidebar link)

### CODE REVIEW

**Comment by s3ththompson** [comment]
cc @lukewagner @titzer @jfbastien @liminzhu

File: demo/index.md [hunk 1]
```diff
@@ -3,34 +3,30 @@ layout: default
...
+  <a class="btn hide-asm-support" href="Tanks/" role="button">Play asm.js fallback</a>
+  <span class="btn-comment btn-comment-error hide-asm-support">Your browser doesn't support WebAssembly yet. <a href="/roadmap/">Learn more</a></span>
 </div>
 
-[![](screenshot.jpg)](AngryBots/)
-<a class="btn btn-primary" id="play-wasm" href="AngryBots/" role="button">Play WebAssembly</a>
-<a class="btn" href="http://beta.unity3d.com/jonas/AngryBots/" role="button">Play asm.js fallback</a>
-
-This is an experimental demo of Angry Bots, a Unity game which has been ported to WebAssembly. Fight robots and explore a 3D space station with realistic environmental effects in this top-down shooter. Movement is controlled by W, A, S, D or the arrow keys and aiming &amp; firing is controlled by the mouse.
+This is a demo of Tanks!, a Unity tutorial game which has been ported to WebAssembly. Drive a tank around the sandbox and shoot the enemy tank. Movement is controlled by W, A, S, D keys and firing is controlled by the spacebar.
```
Comment (lukewagner) [inline demo/index.md]: Could you link "Tanks! a Unity tutorial game" to https://unity3d.com/learn/tutorials/projects/tanks-tutorial?  And then might be good to expand "ported to WebAssembly" to "automatically ported to WebAssembly using the experimental WebGL export"

File: docs/roadmap/index.html [hunk 2]
```diff
@@ -69,28 +63,29 @@ <h2 id="next-steps">Next Steps</h2>
...
 <ul>
-  <li>At the end of the Browser Preview period, the
-	<a href="/docs/binary-encoding/#high-level-structure">binary version</a>
-	will be reset to <code class="highlighter-rouge">0x1</code> (at which point the version will be frozen and all new features will be added in a backwards-compatible manner, and be detected through feature-testing).</li>
+  <li>The <a href="/docs/binary-encoding/#high-level-structure">binary version</a>
+  of WebAssembly has been frozen at <code class="highlighter-rouge">0x1</code> (from this point forward
+  all new features will be added in a backwards-compatible manner,
+  and be detected through feature-testing). To update your
+  WebAssembly modules, upgrade to the latest version of Emscripten 
+  and recompile.</li>
```
Comment (jfbastien) [inline docs/roadmap/index.html]: Is Emscripten the only way to generate WebAssembly binaries that we want to mention?

Is there an easy way to diagnose "obsolete" binaries (e.g. the error messages, etc). As suggested in #1006 it would be good to know which versions of which browser support `0x1` as well, this would ease the transition.

Which version of Emscripten? I believe it's 1.37.3.

File: demo/AngryBots/index.html [hunk 3]
```diff
@@ -1,124 +1,13 @@
...
-    version = String(Wasm.experimentalVersion);
-  } else if (typeof WebAssembly === 'object') {
-    if (WebAssembly.validate(Uint8Array.of(0x0, 0x61, 0x73, 0x6d, 0x01, 0x00, 0x00, 0x00))) {
-        version = String(0x1);
-    } else if (WebAssembly.validate(Uint8Array.of(0x0, 0x61, 0x73, 0x6d, 0x0c, 0x00, 0x00, 0x00))) {
-        version = String(0xc);
-    } else if (WebAssembly.validate(Uint8Array.of(0x0, 0x61, 0x73, 0x6d, 0x0d, 0x00, 0x00, 0x00))) {
-        version = String(0xd);
-    }
-  }
```
Comment (jfbastien) [inline demo/AngryBots/index.html]: Are we losing the feature-tested older versions? I'd at least note this in the PR  description.

File: demo/index.md [hunk 4]
```diff
@@ -3,34 +3,30 @@ layout: default
 markdown: kramdown
 permalink: demo/
 ---
-# Angry Bots Demo
-<div markdown="1" id="wasm-fail" class="flash flash-error flash-hide">
-  Uh-oh! WebAssembly isn't enabled in this browser. To get an early preview of this experimental technology, at your own risk:
```
Comment (jfbastien) [inline demo/index.md]: Can you post a screenshot of what it looks like when wasm isn't enabled?

File: demo/index.md (hunk 1 above)
Comment (jfbastien) [inline demo/index.md]: I'd just drop the word "ported" if Unity supports WebAssembly.

File: demo/AngryBots/index.html (hunk 3 above)
Comment (s3ththompson) [inline demo/AngryBots/index.html]: yes, comment updated

File: demo/index.md (hunk 4 above)
Comment (s3ththompson) [inline demo/index.md]: see screenshot in PR description.

Here's what it looks like when wasm is enabled:

screen shot 2017-03-06 at 11 37 39 am

File: docs/roadmap/index.html (hunk 2 above)
Comment (s3ththompson) [inline docs/roadmap/index.html]: I added a link to the advanced tools page in the message above. Happy to add more tools to that page as stuff comes online (like the gcc backend).

I think the roadmap page isn't the best place for the transition and #1006 stuff... not sure where would be better. Maybe a FAQ under getting started?

I think 0x1 is in 1.37.3, but I'll hold off mentioning a specific version until the async-by-default changes get tagged.

**Comment by s3ththompson** [comment]
Addressed feedback. PTAL, @jfbastien

File: demo/AngryBots/index.html (hunk 3 above)
Comment (jfbastien) [inline demo/AngryBots/index.html]: It may be worth still detecting `0xd` and friends, and popping an error message saying the browser needs to be updated?

File: docs/roadmap/index.html (hunk 2 above)
Comment (jfbastien) [inline docs/roadmap/index.html]: Somewhere else sounds good, we're getting questions about it so it needs to be somewhere.

Can you do this in a separate PR?

Comment (s3ththompson) [inline docs/roadmap/index.html]: Will do

File: demo/AngryBots/index.html (hunk 3 above)
Comment (s3ththompson) [inline demo/AngryBots/index.html]: Will open an issue. Still seems like a burden to simple detection logic (that will be copy-pasted) given the only browsers it applies to should be completely auto-updating...
